### PR TITLE
[SPARK-43988][INFRA] Add a  daily maven testing GitHub Action job

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
TBD
 
### Why are the changes needed?
TBD

### Does this PR introduce _any_ user-facing change?
No, just for test.

### How was this patch tested?
- Pass GitHub Action

### CODE REVIEW

**Comment by LuciferYang** [comment]
wait https://github.com/apache/spark/pull/41487

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -728,6 +729,83 @@ jobs:
...
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: ${{ matrix.java }}
+      - name: Build and Test with Maven
+        shell: 'script -q -e -c "bash {0}"'
+        run: |
+          # Fix for TTY related issues when launching the Ammonite REPL in tests.
+          export TERM=vt100 && script -qfc 'echo exit | amm -s' && rm typescript
+          # `set -e` to make the exit status as expected due to use `script -q -e -c` to run the commands
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:783]: another way is add a new script, maybe named `dev/run-connect-maven-tests` as follows:

```shell
#!/usr/bin/env bash

set -e


# Go to the Spark project root directory
FWDIR="$(cd "`dirname "$0"`"/..; pwd)"
cd "$FWDIR"
export SPARK_HOME=$FWDIR
echo "$SPARK_HOME"


if [[ -z "$JAVA_VERSION" ]]; then
  JAVA_VERSION=8
fi


export MAVEN_OPTS="-Xss64m -Xmx2g -XX:ReservedCodeCacheSize=1g -Dorg.slf4j.simpleLogger.defaultLogLevel=WARN"
export MAVEN_CLI_OPTS="--no-transfer-progress"


# 1. Test with -Phive
# It uses Maven's 'install' intentionally, see https://github.com/apache/spark/pull/26414.
build/mvn $MAVEN_CLI_OPTS -DskipTests -Djava.version=${JAVA_VERSION/-ea} install -Phive
build/mvn $MAVEN_CLI_OPTS -Djava.version=${JAVA_VERSION/-ea} test -pl connector/connect/client/jvm -Phive

# 2. Test without -Phive
build/mvn $MAVEN_CLI_OPTS -DskipTests -Djava.version=${JAVA_VERSION/-ea} install -pl assembly
build/mvn $MAVEN_CLI_OPTS -Djava.version=${JAVA_VERSION/-ea} test -pl connector/connect/client/jvm
```

and `run:` can change to:

```
          # Fix for TTY related issues when launching the Ammonite REPL in tests.
          export TERM=vt100 && script -qfc 'echo exit | amm -s' && rm typescript
          export JAVA_VERSION=${{ matrix.java }}
          ./dev/run-connect-maven-tests
          TEST_RETCODE=$?
          rm -rf ~/.m2/repository/org/apache/spark
          exit $TEST_RETCODE
```

I tested it and it worked, but due to lack of this script in branch-3.4, we need to add `"connect-maven" : "false"`  to `build_branch34.yml`  when using script

**Comment by LuciferYang** [comment]
[7c0409b](https://github.com/apache/spark/pull/41529/commits/7c0409b0a27f63f25974c1e90f252ce8daee62e7) rebase to test

**Comment by LuciferYang** [comment]
Do you think this is necessary? 

For the `connect server` module, testing can be added after resolving `SPARK-43646`

**Comment by LuciferYang** [comment]
friendly ping @HyukjinKwon @dongjoon-hyun

**Comment by LuciferYang** [comment]
also cc @hvanhovell , I would like to add a maven test GA task for the connect modules because I think sbt testing always masks some issues(due to the visibility of the classpath)

**Comment by LuciferYang** [comment]
> Don't feel strongly but I think we should better set a regular job ... with a broader scope .. doing this for every PR is I think a bit of overhead.

For example, a daily job and  tests all modules using maven?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -100,6 +100,7 @@ jobs:
               \"lint\" : \"true\",
               \"k8s-integration-tests\" : \"true\",
               \"breaking-changes-buf\" : \"true\",
+              \"maven-build\" : \"true\",
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:103]: will change to false after test

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -267,6 +268,165 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        java:
+          - ${{ inputs.java }}
+        hadoop:
+          - ${{ inputs.hadoop }}
+        hive:
+          - hive2.3
+        modules:
```
Comment (dongjoon-hyun) [inline .github/workflows/build_and_test.yml:286]: Do you use `modules` in this PR?

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -267,6 +268,165 @@ jobs:
...
+          # Fix for TTY related issues when launching the Ammonite REPL in tests.
+          export TERM=vt100 && script -qfc 'echo exit | amm -s' && rm typescript
+          # `set -e` to make the exit status as expected due to use `script -q -e -c` to run the commands
+          set -e
+          export MAVEN_OPTS="-Xss64m -Xmx4g -Xms4g -XX:ReservedCodeCacheSize=128m -Dorg.slf4j.simpleLogger.defaultLogLevel=WARN"
+          export MAVEN_CLI_OPTS="--no-transfer-progress"
+          export JAVA_VERSION=${{ matrix.java }}
+          ./build/mvn $MAVEN_CLI_OPTS -DskipTests -Pyarn -Pmesos -Pkubernetes -Pvolcano -Phive -Phive-thriftserver -Phadoop-cloud -Djava.version=${JAVA_VERSION/-ea} clean install
+          if [[ "$INCLUDED_TAGS" != "" ]]; then
+            ./build/mvn $MAVEN_CLI_OPTS -pl "$MODULES_TO_TEST" -Pyarn -Pmesos -Pkubernetes -Pvolcano -Phive -Phive-thriftserver -Phadoop-cloud -Djava.version=${JAVA_VERSION/-ea} -Dtest.include.tags="$INCLUDED_TAGS" test
```
Comment (dongjoon-hyun) [inline .github/workflows/build_and_test.yml:405]: Hmm. Got it. It use module. May I ask which this is "for connect client module"?

Comment (dongjoon-hyun) [reply .github/workflows/build_and_test.yml:405]: `resource-managers/yarn,resource-managers/mesos,resource-managers/kubernetes` is irrelevent to this, isn't it?

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:405]: Yes, the topic of this pr will be changed and a daily maven test will be added to cover all modules.

Due to ongoing testing, the pr title and description have not been changed, which has caused you misunderstandings. Sorry for any inconvenience caused @dongjoon-hyun

File: .github/workflows/build_and_test.yml (hunk 3 above)
Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:286]: Yes

**Comment by LuciferYang** [comment]
`ProductAggSuite` in `catalyst` module test aborted when using maven:

```
ProductAggSuite:
*** RUN ABORTED ***
  java.lang.NoClassDefFoundError: Could not initialize class org.apache.spark.sql.catalyst.expressions.codegen.CodeGenerator$
  at org.apache.spark.sql.catalyst.expressions.codegen.JavaCode$.variable(javaCode.scala:64)
  at org.apache.spark.sql.catalyst.expressions.codegen.JavaCode$.isNullVariable(javaCode.scala:77)
  at org.apache.spark.sql.catalyst.expressions.Expression.$anonfun$genCode$3(Expression.scala:200)
  at scala.Option.getOrElse(Option.scala:189)
  at org.apache.spark.sql.catalyst.expressions.Expression.genCode(Expression.scala:196)
  at org.apache.spark.sql.catalyst.expressions.codegen.GenerateSafeProjection$.$anonfun$create$1(GenerateSafeProjection.scala:156)
  at scala.collection.immutable.List.map(List.scala:293)
  at org.apache.spark.sql.catalyst.expressions.codegen.GenerateSafeProjection$.create(GenerateSafeProjection.scala:153)
  at org.apache.spark.sql.catalyst.expressions.codegen.GenerateSafeProjection$.create(GenerateSafeProjection.scala:39)
  at org.apache.spark.sql.catalyst.expressions.codegen.CodeGenerator.generate(CodeGenerator.scala:1369)
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala [hunk 5]
```diff
@@ -46,7 +46,7 @@ import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.Platform
 import org.apache.spark.unsafe.types._
-import org.apache.spark.util.{LongAccumulator, NonFateSharingCache, ParentClassLoader, Utils}
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala:49]: It is possible that SPARK-43300 caused the test aborted described in SPARK-44064, let's revert  SPARK-43300 to confirm

**Pull request closed without merging**